docs/Eclipse4_Migration.md

```diff
@@ -0,0 +1,821 @@
+# Eclipse 4 Migration Guide
+
+## Introduction
+
+This guide helps you migrate your Eclipse 3.x (E3) RCP application to the Eclipse 4 (E4) application model. It assumes you have already set up a basic E4 application structure and now need guidance on how to gradually upgrade existing E3 components to their E4 counterparts.
+
+For a broader understanding of E4 concepts, see the [Eclipse4 RCP FAQ](Eclipse4_RCP_FAQ.md).
+
+## Prerequisites
+
+Before you begin migrating individual components, ensure you have:
+
+1. **A Model Fragment**: All E4 contributions are made through model fragments. If you haven't created one yet, see [Creating Model Fragments](#creating-model-fragments).
+
+2. **Required Dependencies**: Your MANIFEST.MF should include appropriate E4 dependencies:
+   ```
+   Require-Bundle: org.eclipse.e4.ui.model.workbench,
+    org.eclipse.e4.ui.workbench,
+    org.eclipse.e4.core.di,
+    org.eclipse.e4.ui.di,
+    org.eclipse.e4.core.contexts
+   ```
+
+3. **Annotation Support**: For dependency injection to work, add to your MANIFEST.MF:
+   ```
+   Import-Package: javax.annotation;version="1.1.0",
+    javax.inject;version="1.0.0"
```

Comment:
```suggestion
   Import-Package: jakarta.annotation;version="1.1.0",
    jakarta.inject;version="1.0.0"
```

We use the jakarta injections these days, must be adjusted at other places as well